pkg/apis/pipeline/v1alpha1/taskrun_validation.go

```diff
@@ -38,9 +38,16 @@ func (ts *TaskRunSpec) Validate() *apis.FieldError {
 		return apis.ErrMissingField("spec")
 	}
 
-	// Check for TaskRef
-	if ts.TaskRef.Name == "" {
-		return apis.ErrMissingField("spec.taskref.name")
+	// can't have both taskRef and taskSpec at the same time
+	if (ts.TaskRef != nil && ts.TaskRef.Name != "") && ts.TaskSpec != nil {
+		fields := []string{"spec.taskref", "spec.taskspec"}
+		return apis.ErrDisallowedFields(fields...)
```

Comment:
This should work: 

`return apis.ErrDisallowedFields("spec.taskref", "spec.taskspec")`